src/ontodia/widgets/node.tsx

```diff
@@ -0,0 +1,142 @@
+            }
+        } */
+        this.toggle = this.toggle.bind(this);
+        this.showInstances = this.showInstances.bind(this);
+    }
+
+    componentWillReceiveProps(nextProps: NodeTreeProps) {
+        const { searchResult } = nextProps;
+        if (searchResult !== this.props.searchResult) {
+            this.setState({ expanded: false });
+        }
+        if (searchResult && searchResult.length !== 0) {
+            this.setState({ expanded: true });
+        }
+        if (this.state.bgColor === 'rgb(190,235,255)') {
+            this.setState({ bgColor: undefined });
+        }
+    }
+
+    private toggle() {
+        this.setState({ expanded: !this.state.expanded });
+    }
+
+    private showInstances() {
+        this.setState({ bgColor: 'rgb(190,235,255)' });
+        this.props.onClassSelected(this.props.node.id);
+    }
+
+    private getDefaultNodeIcon(): string {
+        if (this.props.node.derived.length !== 0) {
+            return 'parent-tree-icon';
+        } else {
+            return 'default-tree-icon';
+        }
+    }
+    private getToggleIcon(): string {
+        if (this.props.node.derived.length === 0) {
+            return undefined;
+        } else {
+            if (this.state.expanded) {
+                return 'expand-node-icon';
+            } else {
+                return 'collapse-node-icon';
+            }
+        }
+    }
+
+    private boldNode(classLabel: string): Boolean {
+        if (Boolean(this.props.searchResult) && this.props.searchResult.length !== 0) {
+            if (classLabel.toUpperCase().indexOf(this.props.searchString.toUpperCase()) !== -1) {
+                return true;
+            }
+            try { // FatClassModel from dbpedia does not contain information about count
+                if (this.props.node.count.toString().indexOf(this.props.searchString.toUpperCase()) !== -1) {
+                    return true;
+                }
+            } catch (err) {
+                // console.error("class.count === undefined. The search for count will be ignored.")
+            }
+        }
+        return false;
+    }
+    render(): React.ReactElement<any> {
+        const { node, searchResult, searchString, lang, onClassSelected } = this.props;
+        const bgColor = this.state.bgColor;
+        let classLabel = formatLocalizedLabel(node.id, node.label, lang);
+        let bold = this.boldNode(classLabel);
+
+        return (
+            <div className='container' role='tree-item'>
+                <div className={this.getToggleIcon()} onClick={this.toggle} />
+
+                <div className={this.state.mapClassIcons && this.state.mapClassIcons[node.id] ?
+                    this.state.mapClassIcons[node.id] : this.getDefaultNodeIcon()} />
+
+                <div className='tree-class' onClick={this.showInstances} draggable={true}
+                    style={{ fontWeight: bold ? 'bold' : 'normal', background: bgColor }}
+                    onDragStart={e => {
```

Comment:
Why not use an anchor element (`<a href={node.id} ... />`) here? This way you don't have to manually implement `onDragStart` handler.

Reply:
If you replace `<div />` with `<a />` and add a link, then clicking on the class will initiate the link. The ban on processing the click will make the class not clickable, therefore it can not be allocated to search for instances. So I just decided to see how you handled onDragStart in ... `ontodia \ src \ ontodia \ widgets \ instanceSearch.thx` and done exactly the same.

Reply:
It's possible to block only default navigation action by calling `e.preventDefault()` on event object.

Reply:
Anchor was added. Removed the unnecessary code.